File: dataset/codenet/Java/Code/s393796709.java
import java.util.*;
import java.io.*;
public class Main {
    long MOD = 1000000007;
    void solve(BufferedReader in) throws Exception {
        int[] xx = toInts(in.readLine());
        int min = Math.min(xx[0], xx[1]);
        int max = Math.max(xx[0], xx[1]);
        long res;
        if(min == max) {
            long f = f(min);
            res = (2*f*f)%MOD;
        } else if(min + 1 == max){
            long f = f(min);
            res = (f*((f*max)%MOD))%MOD;
        } else {
            res = 0;
        }
        System.out.println(res);
    }
    long f(int n) {
        long res = 1;
        for(int i = 2; i<=n; i++) res = (res*i)%MOD;
        return res;
    }
    int toInt(String s) {return Integer.parseInt(s);}
    int[] toInts(String s) {
        String[] a = s.split(" ");
        int[] o = new int[a.length];
        for(int i = 0; i<a.length; i++) o[i] = toInt(a[i]);
        return o;
    }
    void e(Object o) {
        System.err.println(o);
    }
    public static void main(String[] args) throws Exception{
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        (new Main()).solve(in);
    }
}

File: output/gpt_4o/codenet/direct_translation/java/python/s393796709.py

MOD = 1000000007

def solve():
    xx = to_ints(input().strip())
    min_val = min(xx[0], xx[1])
    max_val = max(xx[0], xx[1])
    if min_val == max_val:
        f = factorial(min_val)
        res = (2 * f * f) % MOD
    elif min_val + 1 == max_val:
        f = factorial(min_val)
        res = (f * ((f * max_val) % MOD)) % MOD
    else:
        res = 0
    print(res)

def factorial(n):
    res = 1
    for i in range(2, n + 1):
        res = (res * i) % MOD
    return res

def to_int(s):
    return int(s)

def to_ints(s):
    a = s.split()
    return [to_int(x) for x in a]

if __name__ == "__main__":
    solve()

